### ISSUE TITLE
Giving error, while Building Resources phase

### ISSUE THREAD

**Issue by okarakose** [shroudedcode/apk-mitm#5]
I am using this command:

`npx apk-mitm app-release.apk`

Error logs are below:

```
Error: Command failed: java -jar /usr/local/lib/node_modules/apk-mitm/jar/apktool.jar build /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode --output /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/unsigned.apk --use-aapt2
W: /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode/res/values/layouts.xml:3: error: invalid value for type 'layout'. Expected a reference.
W: /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode/res/values/public.xml:6060: error: resource 'drawable/$avd_hide_password__0' has invalid entry name '$avd_hide_password__0'. Invalid character '$avd_hide_password__0'.
W: /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode/res/values/public.xml:6061: error: resource 'drawable/$avd_hide_password__1' has invalid entry name '$avd_hide_password__1'. Invalid character '$avd_hide_password__1'.
W: /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode/res/values/public.xml:6062: error: resource 'drawable/$avd_hide_password__2' has invalid entry name '$avd_hide_password__2'. Invalid character '$avd_hide_password__2'.
W: /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode/res/values/public.xml:6063: error: resource 'drawable/$avd_show_password__0' has invalid entry name '$avd_show_password__0'. Invalid character '$avd_show_password__0'.
W: /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode/res/values/public.xml:6064: error: resource 'drawable/$avd_show_password__1' has invalid entry name '$avd_show_password__1'. Invalid character '$avd_show_password__1'.
W: /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode/res/values/public.xml:6065: error: resource 'drawable/$avd_show_password__2' has invalid entry name '$avd_show_password__2'. Invalid character '$avd_show_password__2'.
W: /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode/res/values/public.xml:6071: error: resource 'drawable/$$fingerprint_dialog_error_to_fp__10__0' has invalid entry name '$$fingerprint_dialog_error_to_fp__10__0'. Invalid character '$$fingerprint_dialog_error_to_fp__10__0'.
W: /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode/res/values/public.xml:6072: error: resource 'drawable/$$fingerprint_dialog_error_to_fp__10__1' has invalid entry name '$$fingerprint_dialog_error_to_fp__10__1'. Invalid character '$$fingerprint_dialog_error_to_fp__10__1'.
W: /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode/res/values/public.xml:6073: error: resource 'drawable/$$fingerprint_dialog_error_to_fp__10__2' has invalid entry name '$$fingerprint_dialog_error_to_fp__10__2'. Invalid character '$$fingerprint_dialog_error_to_fp__10__2'.
W: /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode/res/values/public.xml:6074: error: resource 'drawable/$$fingerprint_dialog_error_to_fp__10__3' has invalid entry name '$$fingerprint_dialog_error_to_fp__10__3'. Invalid character '$$fingerprint_dialog_error_to_fp__10__3'.
W: /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode/res/values/public.xml:6075: error: resource 'drawable/$$fingerprint_dialog_error_to_fp__11__0' has invalid entry name '$$fingerprint_dialog_error_to_fp__11__0'. Invalid character '$$fingerprint_dialog_error_to_fp__11__0'.
W: /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode/res/values/public.xml:6076: error: resource 'drawable/$$fingerprint_dialog_error_to_fp__11__1' has invalid entry name '$$fingerprint_dialog_error_to_fp__11__1'. Invalid character '$$fingerprint_dialog_error_to_fp__11__1'.
W: /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode/res/values/public.xml:6077: error: resource 'drawable/$$fingerprint_dialog_error_to_fp__12__0' has invalid entry name '$$fingerprint_dialog_error_to_fp__12__0'. Invalid character '$$fingerprint_dialog_error_to_fp__12__0'.
W: /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode/res/values/public.xml:6078: error: resource 'drawable/$$fingerprint_dialog_error_to_fp__12__1' has invalid entry name '$$fingerprint_dialog_error_to_fp__12__1'. Invalid character '$$fingerprint_dialog_error_to_fp__12__1'.
W: /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode/res/values/public.xml:6079: error: resource 'drawable/$$fingerprint_dialog_error_to_fp__12__2' has invalid entry name '$$fingerprint_dialog_error_to_fp__12__2'. Invalid character '$$fingerprint_dialog_error_to_fp__12__2'.
W: /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode/res/values/public.xml:6080: error: resource 'drawable/$$fingerprint_dialog_error_to_fp__12__3' has invalid entry name '$$fingerprint_dialog_error_to_fp__12__3'. Invalid character '$$fingerprint_dialog_error_to_fp__12__3'.
W: /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode/res/values/public.xml:6081: error: resource 'drawable/$$fingerprint_dialog_error_to_fp__1__0' has invalid entry name '$$fingerprint_dialog_error_to_fp__1__0'. Invalid character '$$fingerprint_dialog_error_to_fp__1__0'.
W: /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode/res/values/public.xml:6082: error: resource 'drawable/$$fingerprint_dialog_error_to_fp__1__1' has invalid entry name '$$fingerprint_dialog_error_to_fp__1__1'. Invalid character '$$fingerprint_dialog_error_to_fp__1__1'.
W: /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode/res/values/public.xml:6083: error: resource 'drawable/$$fingerprint_dialog_error_to_fp__2__0' has invalid entry name '$$fingerprint_dialog_error_to_fp__2__0'. Invalid character '$$fingerprint_dialog_error_to_fp__2__0'.
brut.androlib.AndrolibException: brut.common.BrutException: could not exec (exit code = 1): [/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/brut_util_Jar_3385594703786753249.tmp, compile, --dir, /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode/res, --legacy, -o, /private/var/folders/1n/70zjlg_92y5dsf_rdgyr6hbh0000gn/T/d6fee2ca20bd6ff3d1ac54ceab1b4597/decode/build/resources.zip]
```

**Comment by shroudedcode** [comment]
Thanks for reporting! I have a few solutions for this problem in mind, but I can't test them without the APK file you used. Would you mind sharing it? If it's on Google Play, the name and version of the app is enough. If it's private, you can send it to me via [email](mailto:niklas@shroudedcode.com) or [Telegram](https://t.me/shroudedcode).

**Comment by jzarca01** [comment]
I got the same error using the latest version of uber Uber_v4.279.10002_apkpure.com.apk

**Comment by shroudedcode** [comment]
@jzarca01 That's odd. I was able to successfully patch that APK using `apk-mitm` v0.3.0:

```
$ apk-mitm uber.apk

  ╭ apk-mitm v0.3.0
  ├ apktool v2.4.1 SNAPSHOT@197d46
  ╰ uber-apk-signer v1.1.0
  
  Using temporary directory:
  /tmp/26c10f4763acf921ef5247e68f368c0e

  ✔ Decoding APK file
  ✔ Modifying app manifest
  ✔ Modifying network security config
  ✔ Disabling certificate pinning
  ✔ Encoding patched APK file
  ✔ Signing patched APK file

   Done!  Patched APK: ./uber-patched.apk
```

Could you make sure you're on the latest version and, in case the error occurs again, send me the logs?

**Comment by jzarca01** [comment]
you're right using the latest version of Apktool it works fine
i didnt know where to find that version v2.4.1 SNAPSHOT@197d46 so I just built it from sources

**Comment by shroudedcode** [comment]
@jzarca01 You wouldn't have needed to build it yourself as the Apktool version that is logged at the top (currently `v2.4.1 SNAPSHOT@197d46` — a version I built myself) is already bundled with the `apk-mitm` package. Updating `apk-mitm` would've been enough.

**Comment by okarakose** [comment]
@shroudedcode  My test apk was Akbank Mobile app which you can find in this link : 

[https://play.google.com/store/apps/details?id=com.akbank.android.apps.akbank_direkt](https://play.google.com/store/apps/details?id=com.akbank.android.apps.akbank_direkt)

Some information :

```
About Apk-Mitm
- apk-mitm : v0.2.2
- apktool : v2.4.1 SNAPSHOT@197d46
- uber-apk-signer : v1.1.0

About Node && Npm
- Node version : 8.12.0
- NPM version : 6.4.1

About Patching Apk
- Akbank mobile version : v4.0.2
```

**Comment by shroudedcode** [comment]
@okarakose It turns out that APK builds perfectly fine when the legacy version of AAPT is used. `apk-mitm` uses AAPT2 by default though which seems to cause issues with some APKs. That's why I've decided to have `apk-mitm` fall back to AAPT when building with AAPT2 fails.

I've just released version [0.4.0](https://github.com/shroudedcode/apk-mitm/releases) with this feature, so you should be able to patch that APK file now. You can run this command to force `npx` to download the new version:

```
npx apk-mitm@^0.4.0 app-release.apk
```

**Comment by okarakose** [comment]
@shroudedcode Yes, you are right. It is worked with 0.4.0 version. Thank you
